operators/pkg/controllers/bastion_controller.go

```diff
@@ -0,0 +1,86 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	crownlabsalpha1 "github.com/netgroup-polito/CrownLabs/operators/api/v1alpha1"
+)
+
+// BastionReconciler reconciles a Bastion object
+type BastionReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=crownlabs.polito.it,resources=tenants,verbs=list
+
+func (r *BastionReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("bastion", req.NamespacedName)
+	log.Info("reconciling bastion")
+
+	// Get tenants resources
+	var tenants crownlabsalpha1.TenantList
+	if err := r.List(ctx, &tenants); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// Collect public keys of tenants
+	keys := make([]string, 0, len(tenants.Items))
+	for _, tenant := range tenants.Items {
+		keys = append(keys, tenant.Spec.PublicKeys...)
+	}
+
+	authorizedKeys := strings.Join(keys[:], "\n")
+
+	f, err := os.Create("/auth-keys-vol/authorized_keys")
+	if err != nil {
+		log.Error(err, "unable to create the file authorized_keys")
+		return ctrl.Result{}, nil
+	}
+	_, err = f.WriteString(authorizedKeys)
+	if err != nil {
+		log.Error(err, "unable to write to authorized_keys")
+		f.Close()
```

Comment:
It is probably better to use `defer closeFile()` right below when you open the file, to ensure it is always closed (where closeFile() is a function which closes the file and checks for errors as you are doing below)